### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by ianlavine**

### CODE REVIEW

File: src/main/java/attributes/AttributeMap.java [hunk 1]
```diff
@@ -29,8 +29,12 @@ public void addItem(String name, Object item) throws ClassCastException {
      *
      * @param name The name of the Attribute to be added to the map
      * @param item The Attribute to be added to the map
+     * @throws NullPointerException if item is null
      */
-    public void addItem(String name, Attribute item) {
+    public void addItem(String name, Attribute item) throws NullPointerException {
```
Comment (mtoohey31) [inline src/main/java/attributes/AttributeMap.java:34]: Do we need to check for this here? We haven't checked for null items anywhere else right? If there's a reason we need these items to be non-null lmk though, I might just have missed it! Was this something IntelliJ was complaining about?

**Review by mtoohey31 — CHANGES_REQUESTED** [review]
Looks great, just one other thing in addition to the review comments, the `.DS_Store` file shouldn't be included in the repository, feel free to add `**/.DS_Store` to the `.gitignore` if git keeps trying to add it though!

File: src/main/java/attributes/AttributeMap.java (hunk 1 above)
Comment (Mogtaba-Alim) [reply src/main/java/attributes/AttributeMap.java:34]: No you're right it isnt something we should be checking for it was part of the old refactored code that was removed.

File: src/main/java/entitypackagers/JsonPackager.java [hunk 2]
```diff
@@ -14,8 +14,9 @@
 public class JsonPackager implements Packager {
 
     /**
-     * Writes packageMap to a JsonObject
+     * Writes packageMap to a JsonObject'
```
[the remark this answers is not in the document]
Comment (Mogtaba-Alim) [reply src/main/java/entitypackagers/JsonPackager.java:17]: Alright

**Review by mtoohey31 — APPROVED** [review]
Thanks for the changes, this is good to go!

**Pull request merged**